src/app/shared/daterange/daterange.component.ts

```diff
@@ -0,0 +1,226 @@
+import { OverlayRef, GlobalPositionStrategy, Overlay } from '@angular/cdk/overlay';
+import { Observable, Observer, CompletionObserver } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { ComponentPortal, PortalInjector } from '@angular/cdk/portal';
+import { DaterangePortalComponent } from './daterange-portal.component';
+import { PORTAL_DATA } from './daterange-injection';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'grove-daterange',
+  template: `
+    <mat-form-field>
+      <input
+        matInput
+        [value]="startDate"
+        [matDatepicker]="fromPicker"
+        (dateInput)="startDateChangeEvent('input', $event)"
+        (dateChange)="startDateChangeEvent('change', $event)"
+        placeholder="Start date"
+        disabled
+      />
+      <mat-datepicker #fromPicker disabled="false"></mat-datepicker>
+    </mat-form-field>
+    <mat-form-field>
+      <input
+        matInput
+        [value]="endDate"
+        [matDatepicker]="toPicker"
+        [matDatepickerFilter]="rangeFilter"
+        (dateInput)="endDateChangeEvent('input', $event)"
+        (dateChange)="endDateChangeEvent('change', $event)"
+        placeholder="End date"
+        disabled
+      />
+      <mat-datepicker [dateClass]="highlightStartDate" [startAt]="toPickerInitialDate" #toPicker disabled="false"></mat-datepicker>
+    </mat-form-field>
+    <button mat-raised-button (click)="openPicker(fromPicker)">Open</button>
+    <grove-daterange-overlay [display]="showBackdrop"></grove-daterange-overlay>
+  `,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DaterangeComponent),
+      multi: true
+    }
+  ],
+  styleUrls: ['daterange.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class DaterangeComponent implements AfterViewInit, OnDestroy, ControlValueAccessor {
+  @ViewChild('fromPicker') fromPicker: MatDatepicker<Date>;
+  @ViewChild('toPicker') toPicker: MatDatepicker<Date>;
+  @ViewChild(MatDatepickerInput) fromInput: MatDatepickerInput<Date>;
+  @Input() _startDate = null;
+  @Input() _endDate = null;
+
+  set startDate(val: Date) {
+    this._startDate = val;
+    this.propagateChange({startDate: this.startDate, endDate: this.endDate});
+  }
+
+  get startDate() {
+    return this._startDate;
+  }
+
+  set endDate(val: Date) {
+    this._endDate = val;
+    this.propagateChange({startDate: this.startDate, endDate: this.endDate});
+  }
+
+  get endDate() {
+    return this._endDate;
+  }
+
+  authToken: string;
+  rafRegistry = {};
+  overlay: Overlay;
+  overlays: OverlayRef[] = [];
+  showBackdrop = false;
+  get toPickerInitialDate() {
+    if (this.startDate) { return this.startDate; }
+    return new Date();
+  }
+
+  propagateChange = (_: any) => {};
+
+  constructor(overlay: Overlay, private _injector: Injector) {
```

Comment:
can just make the overlay arg private and dont have to declare as a member or assign it